reactivesocket-core/src/main/java/io/reactivesocket/util/PayloadImpl.java

```diff
@@ -61,17 +71,39 @@ public PayloadImpl(ByteBuffer data) {
     }
 
     public PayloadImpl(ByteBuffer data, ByteBuffer metadata) {
+        this(data, metadata, false);
+    }
+
+    /**
+     * New instance where every invocation of {@link #getMetadata()} and {@link #getData()} will reset the position of
+     * the buffer to the position when it is created, if and only if, {@code singleUse} is set to {@code false}.
+     *
+     * @param data Buffer for data.
+     * @param metadata Buffer for metadata.
+     * @param singleUse {@code true} if the buffer position is to be reset on every invocation of {@link #getData()} and
+     * {@link #getMetadata()}.
+     */
+    public PayloadImpl(ByteBuffer data, ByteBuffer metadata, boolean singleUse) {
         this.data = data;
```

Comment:
@robertroeser `ByteBuffer.slice()` always uses the current position i.e. there is no such method as `ByteBuffer.slice(position)`. So, if we were to slice, we still have to reset the position first.
Do you see any other value in using `slice()` even though we have to reset position first?